### DISCUSSION TITLE
User permissions not revoked in laravel/dusk browser scope

### DISCUSSION THREAD

**Discussion by dmvpformation** [spatie/laravel-permission#2128] [General]
**Describe the bug**
We encountered a bug while testing permissions access with laravel/dusk and your package.

It's quite basic, we revoke permissions of a user, login and then close a tutorial modal. The problem is that when we check the permission that is supposed to be revoked in the dusk process, the user still has this permission. Though, it is not listed when printing permissions via `getPermissionNames` on both Role and User objects. We checked that it is revoked before starting to browse and it is.

Our guess is that there is a different scope or some kind of cache that still gives this permission to our user, but we cannot figure out where or how.

The behavior of the bug is pretty erratic. It is not consistent on every environment, plus it disappeared on some environments after an update of your package to version 5.5.4. However, it is consistent in our gitlab-ci, it keeps triggering on this precise unique test.

SetUp of dusk test:
```
    public function setUp(): void
    {
        // first include all the normal setUp operations
        parent::setUp();

        // now re-register all the roles and permissions (clears cache and reloads relations)
        $this->app->make(PermissionRegistrar::class)->registerPermissions();
        

    }
```

Permissions revoking:
```
        /**
         *  CREATE DEFAULT DATABASE CONTENT
         */
        Role::findByName(Roles::ROLE_1)->revokePermissionTo(Permissions::ACCESS_ADMIN, Permissions::ACCESS_RESSOURCES);
        $this->userRole1->revokePermissionTo(Permissions::ACCESS_ADMIN);
```

Dusk test browse:
```
        $this->browse(function (BaseBrowser $browser) {
            $browser->login($this->userRole1->email)
                // CANNOT FIND MODAL BECAUSE OF TOO MANY PERMISSIONS
                ->closeModal();
        }
```

**Versions**
You can use `composer show` to get the version numbers of:
- spatie/laravel-permission package version: 5.5.4
- laravel/dusk: 6.23.0

**Expected behavior**
We expect the permissions to be revoked and stay revoked inside the browser test.

**Environment (please complete the following information, because it helps us investigate better):**
 - Gitlab CI, Gitlab version : 15.0
 - Ubuntu: 22.04 LTS

**Comment by freekmurze** [comment]
I currently don't have the time to investigate this further.

Feel free to submit a PR that fixes the problem for you.

**Reply by dmvpformation** [reply]
Hello,

We currently don't have a solution. In order to bypass this on our CI, we did something we're not really proud of for now: We moved the dusk test in the first place and it doesn't trigger the bug above. We will get back to this later but for now we'll have to do with this "band aid".

We know it's not recommended nor a solution per se, but again, we cannot spend much more time on this than we already did.

If we ever find a solution, we'll keep you updated.
